### PR TITLE
refactor: update type guards to use `in` where reasonable

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: src/transform.ts [hunk 1]
```diff
@@ -609,50 +609,50 @@ export interface LoessTransform {
...
-  return t['flatten'] !== undefined;
+  return 'flatten' in t;
 }
 export function isCalculate(t: Transform): t is CalculateTransform {
-  return t['calculate'] !== undefined;
+  return 'calculate' in t;
 }
 
 export function isBin(t: Transform): t is BinTransform {
-  return !!t['bin'];
```
Comment (domoritz) [inline src/transform.ts]: What if `bin: false`? We should not change this one here.

Comment (kanitw) [reply src/transform.ts]: bin's type is `true | BinParams`. 

If we worry about falsy for the primary properties, then maybe we shouldn't change this for all transforms?

For example, 

`{aggregate: null/undefined}` is probably not a reasonable AggregateTransform?

Comment (domoritz) [reply src/transform.ts]: Bin is the only one I'm really worried about since people may expect that `false` works since we support `true`.

Comment (kanitw) [reply src/transform.ts]: But why would you do that in transform? 
If so, does that make it a bin transform or not? Perhaps an inactive bin transform? So “in” is still correct.

Comment (domoritz) [reply src/transform.ts]: Ohh, nm, I looked wrong.

**Comment by domoritz** [comment]
Runtime tests are still failing.

**Pull request merged**